### PR TITLE
deprecated: epoch indexing from 1

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Clean up deprecated epoch indexing, starting from 1

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
shall we also index batches from 1? now it is a bit confusing that one is indexed from 1 other from 0
cc: @williamFalcon @jeremyjordan @PyTorchLightning/core-contributors

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Raise an error when lightning replaces an existing sampler ([#2020](https://github.com/PyTorchLightning/pytorch-lightning/pull/2020))
 - Enabled prepare_data from correct processes - clarify local vs global rank ([#2166](https://github.com/PyTorchLightning/pytorch-lightning/pull/2166))
 - Remove explicit flush from tensorboard logger ([#2126](https://github.com/PyTorchLightning/pytorch-lightning/pull/2126))
+- Changed epoch/step indexing from 1 instead of 0 ([#2206](https://github.com/PyTorchLightning/pytorch-lightning/pull/2206))
```
Comment (awaelchli) [inline CHANGELOG.md:51]: now the index starts from 0 instead of 1, right?

File: tests/models/test_hooks.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def training_epoch_end(self, outputs):
     # a metric shared in both methods gets overwritten by epoch_end
     assert metrics['shared_metric'] == 111
     # metrics are kept after each epoch
-    for i in range(num_epochs):
+    for i in range(1, num_epochs + 1):
```
Comment (awaelchli) [inline tests/models/test_hooks.py:71]: why is it starting from 1 here?

File: CHANGELOG.md (hunk 1 above)
Comment (Borda) [reply CHANGELOG.md:51]: no now it starts from 1

File: tests/models/test_hooks.py (hunk 2 above)
Comment (Borda) [reply tests/models/test_hooks.py:71]: now it starts from 1

**Pull request merged**